src/libsyntax/ast.rs

```diff
@@ -399,6 +399,16 @@ impl Generics {
     pub fn is_parameterized(&self) -> bool {
         self.is_lt_parameterized() || self.is_type_parameterized()
     }
+    pub fn span_for_name(&self, name: &str) -> Option<Span> {
```

Comment:
Seems okay.  If there are duplicate type parameter names, you'll get a different error for that.